packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -327,13 +334,39 @@ abstract class BindingBase {
     registerServiceExtension(
       name: name,
       callback: (Map<String, String> parameters) async {
-        if (parameters.containsKey(name))
+        if (parameters.containsKey(name)) {
           await setter(double.parse(parameters[name]));
+          _postExtensionChangedEvent(name, (await getter()).toString());
+        }
         return <String, dynamic>{ name: (await getter()).toString() };
       }
     );
   }
 
+  /// Sends an event when a service extension's state is changed. Clients should
+  /// listen for this event to stay aware of the current service extension
+  /// state. Any service extension that manages a state should call this method
+  /// on state change.
+  ///
+  /// `value` reflects the newly updated service extension value.
+  void _postExtensionChangedEvent(String name, dynamic value) {
```

Comment:
Should this be more accurately named `postExtensionStateChangedEvent` since it's not really the extension that's changing?

Reply:
Done.